plugin/trino-hive/src/main/java/io/trino/plugin/hive/HivePageSink.java

```diff
@@ -338,6 +357,27 @@ private void closeWriter(int writerIndex)
         partitionUpdates.add(wrappedBuffer(partitionUpdateCodec.toJsonBytes(partitionUpdate)));
     }
 
+    @Override
+    public void closeIdleWriters(long estimatedWrittenBytes, long writtenBytesThreshold)
```

Comment:
`void closeIdleWriters()`

then `TableWriterOperator` based on `physicalWrittenBytesThreshold` (over all drivers) will call `closeIdleWriters()`

close writer if it didn't move since last `closeIdleWriters`